File: Java/B/BinSearch_Recursive.java
class Main {
	public static int binarySearch(int[] numerics, int target, int lowerIndex, int upperIndex) {
		if(upperIndex < lowerIndex) {
			return -1;
		}

		int midIndex = (lowerIndex + upperIndex) / 2;

		if (numerics[midIndex] > target) {
			return binarySearch(numerics, target, lowerIndex, midIndex - 1);
		} 
		else {
			return binarySearch(numerics, target, midIndex + 1, upperIndex);
		}
	}
	
	public static void main(String[] args) {
		int[] arr = {1, 3, 5, 6, 9, 10};
		int target = 0;
		int result = binarySearch(arr, target, 0, arr.length);

		if (result == -1) {
			System.out.println(target + " was not found in the array!");
		} 
		else {
			System.out.println(target + " was found in array element " + result);
		}

	}
}

File: Python/B/BinSearch_Recursive.py
def binSearch(arr, lowerIndex, upperIndex, target):
	if upperIndex >= lowerIndex:
		midIndex = (lowerIndex + upperIndex) // 2

		if arr[midIndex] == target:
			return midIndex
		elif arr[midIndex] > target:
			return binSearch(arr, lowerIndex, midIndex - 1, target)
		else:
			return binSearch(arr, midIndex + 1, upperIndex, target)
	
	else:
		return -1

arr = [1, 2, 3, 4, 5]
target = 5
result = binSearch(arr, 0, len(arr) - 1, target)

if result != -1:
	print("Element is present in array element: ", str(result))
else:
	print("Element is not present in the array!")
